grafana/grafana
# Prometheus: Handle the response with different field key order

**What is this feature?**

For some edge cases prometheus response is returning with out of order field keys. We must read the `resultType` before the `result` while parsing the data. 
See the example http response here https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries
Since that is an edge case this PR is covering by re-iterating the result after reading the `resultType`

**Why do we need this feature?**

To be able to cover out of order field key responses.

**Who is this feature for?**

Potentially all prometheus data source users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/64693

**Special notes for your reviewer:**

#### How to test
- You can modify the prometheus response as it will return the `resultType` after `result` field
- Or you can just run the unit tests I provided in the `main` branch and this branch.

Please check that:
